### PR TITLE
update(mini-css-extract-plugin): v1 updates

### PR DESCRIPTION
**Pull request by peterblazejewicz**
- `esModule` now defaults to true
- `filename` option as function
- `moduleFilename` removed
- version bump
- tests amended

https://github.com/webpack-contrib/mini-css-extract-plugin/compare/v0.12.0...v1.0.0

Thanks!

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)

### CODE REVIEW

File: types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts [hunk 1]
```diff
@@ -49,14 +49,14 @@ configuration = {
 
 configuration = {
     // ...
-    plugins: [new MiniCssExtractPlugin()],
+    plugins: [new MiniCssExtractPlugin({})],
```
Comment (skovy) [inline types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts:52]: Why this change? If you aren't configuring anything shouldn't `MiniCssExtractPlugin()` be valid?

File: types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts [hunk 2]
```diff
@@ -49,14 +49,14 @@ configuration = {
...
 };
 
 configuration = {
     // ...
     plugins: [
         new MiniCssExtractPlugin({
-            filename: 'styles.css',
+            filename: ({ chunk }) => `${chunk.name.replace('/js/', '/css/')}.css`,
```
Comment (skovy) [inline types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts:59]: There is now no longer a test with `filename` with a value of `string`? Probably would be good to test both `string` and the function format?

Comment (peterblazejewicz) [reply types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts:59]: no, there are in other blocks before and after, e.g:
https://github.com/peterblazejewicz/DefinitelyTyped/blob/update/mini-css-extract-plugin/types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts#L69

File: types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts:52]: oh, this is just kind of typo, I was under impression the parameterless ctro is already ther in thest,
to be amended. thx!

Comment (peterblazejewicz) [inline types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts:52]:
```suggestion
    plugins: [new MiniCssExtractPlugin(), new MiniCssExtractPlugin({})],
```

File: types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts [hunk 3]
```diff
@@ -49,14 +49,14 @@ configuration = {
 
 configuration = {
     // ...
-    plugins: [new MiniCssExtractPlugin()],
+    plugins: [new MiniCssExtractPlugin(), new MiniCssExtractPlugin({})],
```
Comment (JounQin) [inline types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts:52]: Not sure why an empty object param should be added here.

File: types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts (hunk 2 above)
Comment (skovy) [reply types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts:59]: ah my bad, I thought I expanded everything in the diff view but missed that, sorry about that!

File: types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts (hunk 3 above)
Comment (peterblazejewicz) [reply types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts:52]: Oh, it's just a test for coding issue. Allowed by package itself, all options are not required and comes with defualts.
Should I remove this? The contract allows user error (object with no properties is fully supported)
running that code is perfectly OK and creates:
```js
MiniCssExtractPlugin {
  options: {
    filename: '[name].css',
    ignoreOrder: false,
    chunkFilename: '[name].css'
  }
}
```

Comment (skovy) [reply types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts:52]: I wasn't sure if it was a valid use case in practice and if not if it was worth testing?

I don't see a problem in keeping it though if you think it's important to have tested 👍

**Review by skovy — APPROVED** [review]

**Comment by peterblazejewicz** [comment]
Ready to merge 💋
